### DISCUSSION TITLE
Transformer based PoS tagger doesn't return PoS tags for pretokenized texts

### DISCUSSION THREAD

**Discussion by ulf1** [explosion/spaCy#9042] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

```
import de_dep_news_trf as spacy_model
import spacy
x_test = [["Erstes", "Dokument", "."], ["spacy3", "will", "nicht", "."]]
model = spacy_model.load()
docs = [model.get_pipe("tagger")(spacy.tokens.doc.Doc(model.vocab, words=sequence))
        for sequence in x_test]
[[t.tag_ for t in doc] for doc in docs]
```

returns
```
[['$(', '$(', '$('], ['$(', '$(', '$(', '$(']]
```

## Your Environment

* Operating System: Ubuntu
* Python Version Used: 3.7
* spaCy Version Used: 3.1.*
* Environment Information: 

The problem is related to https://github.com/explosion/spaCy/issues/7105

**Comment by adrianeboyd** [comment, marked as answer]
Hi, this is the same issue as in #7105, except that in this pipeline it's the `transformer` component that needs to run in addition to `tagger` instead of `tok2vec`+`tagger`. This should work:

```python
doc = nlp.get_pipe("transformer")(doc)
doc = nlp.get_pipe("tagger")(doc)
```

The tagger "listens" to the transformer component so it doesn't have the right input unless the transformer component runs first, which is why you get a bunch of nonsense tags as output in your example above. There are some more details about the pretrained pipeline design here: https://spacy.io/models#design
